### DISCUSSION TITLE
Can I change a color of a single mesh in model?

### DISCUSSION THREAD

**Discussion by KratkiSebaw** [google/model-viewer#3544] [Q&A] [answered]
I would like to achieve the effect like in the pictures

<img src="https://i.stack.imgur.com/cL64Y.png"/>
<img src="https://i.stack.imgur.com/1xrIV.png"/>

I have already tried with materials API, but to no avail. 
Because I don't want to change the color of the material, but a single object.

Is it possible?

**Comment by elalish** [comment, marked as answer]
Our API only has the granularity of materials. This means you need to author the file such that each object that you want to change has its own material.

**Reply by DanieleMagri** [reply]
But within the code, how do you tell model-viewer to change the texture\material **only** of that object?

**Reply by elalish** [reply]
So, for instance, name the material you've assigned to that object "bob". Then use:
```
const material = mv.model.getMaterialByName('bob');
const texture = mv.createTexture('path/to/image.jpg');
material.pbrMetallicRoughness.baseColorTexture.setTexture(texture);
```
https://modelviewer.dev/examples/scenegraph/#pickMaterialExample
You can also get it by glTF index: `mv.model.materials[i];`

**Reply by ShepDeCuisine** [reply]
Hey, i'm looking to have a shirt and hat mesh that I can change the solid colour on then also have the ability to change a graphic that would be a png with alpha. I want the user to be able to pick from different colours and different graphics to create their custom look - any chance you could point me in the right direction to achieve this?
